k9mail/src/main/java/com/fsck/k9/activity/setup/AccountSetup.java

```diff
@@ -120,6 +174,52 @@ private void setSelection(int position) {
 
         this.position = position;
         flipper.setSelection(position);
+
+        switch (layoutIds[position]) {
+            case R.layout.account_setup_basics:
+                basicsView = new BasicsView(this);
+                basicsView.start();
+                break;
+            case R.layout.account_setup_autoconfiguration:
+                checkSettingsView = new CheckSettingsView(this);
+                checkSettingsView.start();
+                break;
+            case R.layout.account_setup_incoming:
+                incomingView = new IncomingView(this);
+                incomingView.start();
+                break;
+            case R.layout.account_setup_account_type:
+                accountTypeView = new AccountTypeView(this);
+                accountTypeView.start();
+                break;
+            case R.layout.account_setup_names:
+                namesView = new NamesView(this);
+                namesView.start();
+                break;
+        }
     }
 
+    @Override
+    public void doPositiveClick(int dialogId) {
+        switch (dialogId) {
+            case R.id.dialog_account_setup_error: {
+                // presenter.onEditDetailClickedWhenError();
```

Comment:
Here is one of the places where I don't know what is the best way. The listener must be an activity. In order to call presenter's methods, should I declare a method in view and then call methods in presenter indirectly?